tests/helpers/parsers.js

```diff
@@ -37,7 +37,7 @@ const parsers = {
   BABEL_ESLINT: path.join(__dirname, NODE_MODULES, 'babel-eslint'),
   '@BABEL_ESLINT': path.join(__dirname, NODE_MODULES, '@babel/eslint-parser'),
   TYPESCRIPT_ESLINT: path.join(__dirname, NODE_MODULES, 'typescript-eslint-parser'),
-  '@TYPESCRIPT_ESLINT': path.join(__dirname, NODE_MODULES, '@typescript-eslint/parser'),
+  '@TYPESCRIPT_ESLINT': path.join(__dirname, NODE_MODULES, '@typescript-eslint/parser/dist'),
```

Comment:
`@typescript-eslint/parser` removed the `main` field in `v6`, which pointed to `dist/index.js`. Not sure how tests were passing before, as this would have been wrong for all of v6.

Reply:
v6 supports node versions `^16.0.0 || >=18.0.0`, but even if I'm running this locally with node 18 it still fails to find the export

Reply:
interesting, indeed node 18 should work - line 15 in https://unpkg.com/browse/@typescript-eslint/parser@6.19.0/package.json points there.

Reply:
any update here?

Reply:
Short version: I added another commit which replaces the path joining with `require.resolve` 🙂 If it needs to be `path.join` there needs to be some edge case handling here.

Longer version: I could get this working using both `/dist` and `dist/index.js` but not without any of those. It kind of makes sense why this doesn't work with newer versions of `@typescript-eslint/parser` (with an exports field) if you look at the [resolution logic node uses](https://nodejs.org/api/modules.html#all-together). It seems like the exports field are only taken into consideration when code is required as node modules, which node doesn't do if the import path is a file system path (starts with `./`, `../` or `/`). If a file system path is provided, it will only try loading it as a file or directory. When loading as a directory only the `main` field is taken into account, not the `exports` field.

So when just providing `path.join(__dirname, NODE_MODULES, '@typescript-eslint/parser` as the path it will get an absolute path to the directory so node will only look for the main field. I tried to find some context behind why the `path.join` pattern was used in the [commit introducing it](https://github.com/jsx-eslint/eslint-plugin-react/blob/9fbd037193d9f5c0980e11e80825e726d61c8ab3/tests/helpers/parsers.js) but could not find a PR or anything. Anyway, replacing those with `require.resolve('<parser lib>')` gives an absolute path but kicks all of the resolving work over to node and makes it go the "node modules route" instead which resolves the exports field 👍 It's supported on all node versions, so no issues in the workflows from using this instead of manually constructing absolute paths 🙂

Reply:
ok that makes sense.